src/CompareRevisions/Kube.hs

```diff
@@ -169,41 +173,24 @@ mapDiff source target = Map.fromList (go (Map.toAscList source) (Map.toAscList t
         GT -> (yKey, Removed yValue):go xs ys'
 
 
--- | Here, an environment is a mapping from Kubernetes to their definitions.
-type Env = Map KubeObject Value
+-- | Here, an environment is collection of Kubernetes objects.
+type Env = [KubeObject]
 
 loadEnvFromDisk :: MonadIO m => FilePath -> m Env
 loadEnvFromDisk directory = do
   files <- getFiles directory
   let yamlFiles = [ f | f <- files, takeExtension f == ".yaml" ]
   bytes <- traverse (liftIO . ByteString.readFile) yamlFiles
-  let values = mapMaybe Yaml.decode bytes -- ignore files that don't parse to yaml
-  pure (Map.fromList (mapMaybe valueToPair values)) -- ignore yaml that doesn't look like kubeobject
-  where
-    valueToPair v =
-      case Aeson.fromJSON v of
-        Aeson.Error _ -> Nothing
-        Aeson.Success kubeObj -> Just (kubeObj, v)
-
--- | Given a @kubeConfig@ and some @kubeObjects@, fetch their definitions from
--- a cluster.
---
--- If anything goes wrong running the @kubectl@ process at any time, just give
--- up.
-loadEnvFromCluster :: Maybe FilePath -> [KubeObject] -> ExceptT ProcessError IO Env
```

Comment:
Deleting this is just opportunistic dead code removal I think?

Reply:
Yes, that's correct. I think this code is a straight up port of the kubediff Python utility that compares local kube state to cluster.